Mathlib/RingTheory/Ideal/Height.lean

```diff
@@ -0,0 +1,118 @@
+/-
+Copyright (c) 2025 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiedong Jiang, Jingting Wang, Andrew Yang
+-/
+import Mathlib.Order.KrullDimension
+import Mathlib.RingTheory.Ideal.MinimalPrime
+import Mathlib.RingTheory.PrimeSpectrum
+/-!
+# The Height of an Ideal
+
+In this file, we defined the height of a prime ideal and the height of an ideal.
+
+# Main definitions
+
+* `Ideal.primeHeight` : The height of a prime ideal $\mathfrak{p}$. We defined it as the length of
+the maximum chain of prime ideals whose maximal element is $\mathfrak{p}$ minus 1.
+
+* `Ideal.height` : The height of an ideal. We defined it as the infimum of the primeHeight of the
+minimal prime ideals containing I.
+
+-/
+
+variable {R : Type*} [CommRing R] (I : Ideal R)
+
+open Ideal
+
+/-- The height of a prime ideal is defined as the supremum of the lengths of strictly decreasing
+chains of prime ideals below it. -/
+noncomputable def Ideal.primeHeight [hI : I.IsPrime] : ℕ∞ :=
+  Order.height (⟨I, hI⟩ : PrimeSpectrum R)
+
+/-- The height of an ideal is defined as the infimum of the heights of minimal prime ideals
+containing it. -/
+noncomputable def Ideal.height : ℕ∞ :=
+  ⨅ J ∈ I.minimalPrimes, @Ideal.primeHeight _ _ J (minimalPrimes_isPrime ‹_›)
+
+/-- For a prime ideal, its height equals its prime height. -/
+lemma Ideal.height_eq_primeHeight [I.IsPrime] : I.height = I.primeHeight := by
+  unfold height primeHeight
+  simp_rw [Ideal.minimalPrimes_eq_subsingleton_self]
+  simp
+
+/-- An ideal has finite height if it is either the unit ideal or its height is finite.
+We include the unit ideal in order to have the instance `IsNoetherianRing R → FiniteHeight I`. -/
+class Ideal.FiniteHeight : Prop where
+  eq_top_or_height_ne_top : I = ⊤ ∨ I.height ≠ ⊤
+
+lemma Ideal.finiteHeight_iff_lt {I : Ideal R} :
+  Ideal.FiniteHeight I ↔ I = ⊤ ∨ I.height < ⊤ := by
```

Comment:
```suggestion
    Ideal.FiniteHeight I ↔ I = ⊤ ∨ I.height < ⊤ := by
```
The statement should be indented by 4 spaces. Same below.

Reply:
Just reminding that the statement should be indented 4 spaces, in case my last message was missed.